webapp/python/app.py

```diff
@@ -0,0 +1,412 @@
+        if width["min"] != -1:
+            conditions.append("width >= %s")
+            params.append(width["min"])
+        if width["max"] != -1:
+            conditions.append("width < %s")
+            params.append(width["max"])
+
+    if args.get("depthRangeId"):
+        for _range in chair_search_condition["depth"]["ranges"]:
+            if _range["id"] == int(args.get("depthRangeId")):
+                depth = _range
+                break
+        else:
+            raise BadRequest("depthRangeId invalid")
+        if depth["min"] != -1:
+            conditions.append("depth >= %s")
+            params.append(depth["min"])
+        if depth["max"] != -1:
+            conditions.append("depth < %s")
+            params.append(depth["max"])
+
+    if args.get("kind"):
+        conditions.append("kind = %s")
+        params.append(args.get("kind"))
+
+    if args.get("color"):
+        conditions.append("color = %s")
+        params.append(args.get("color"))
+
+    if args.get("features"):
+        for feature_confition in args.get("features").split(","):
+            conditions.append("features LIKE CONCAT('%', %s, '%')")
+            params.append(feature_confition)
+
+    if len(conditions) == 0:
+        raise BadRequest("Search condition not found")
+
+    conditions.append("stock > 0")
+
+    try:
+        page = int(args.get("page"))
+    except (TypeError, ValueError):
+        raise BadRequest("Invalid format page parameter")
+
+    try:
+        per_page = int(args.get("perPage"))
+    except (TypeError, ValueError):
+        raise BadRequest("Invalid format perPage parameter")
+
+    search_condition = " AND ".join(conditions)
+
+    query = f"SELECT COUNT(*) as count FROM chair WHERE {search_condition}"
+    count = select_row(query, params)["count"]
+
+    query = f"SELECT * FROM chair WHERE {search_condition} ORDER BY popularity DESC, id ASC LIMIT %s OFFSET %s"
+    chairs = select_all(query, params + [per_page, per_page * page])
+
+    return {"count": count, "chairs": camelize(chairs)}
+
+
+@app.route("/api/chair/search/condition", methods=["GET"])
+def get_chair_search_condition():
+    return chair_search_condition
+
+
+@app.route("/api/chair/<int:chair_id>", methods=["GET"])
+def get_chair(chair_id):
+    chair = select_row("SELECT * FROM chair WHERE id = %s", (chair_id,))
+    if chair is None or chair["stock"] <= 0:
+        raise NotFound()
+    return camelize(chair)
+
+
+@app.route("/api/chair/buy/<int:chair_id>", methods=["POST"])
+def post_chair_buy(chair_id):
+    cnx = mysql.connector.connect(**mysql_connection_env)
+    try:
+        cnx.start_transaction()
+        cur = cnx.cursor(dictionary=True)
+        cur.execute("SELECT * FROM chair WHERE id = %s AND stock > 0 FOR UPDATE", (chair_id,))
+        chair = cur.fetchone()
+        if chair is None:
+            raise NotFound()
+        cur.execute("UPDATE chair SET stock = %s WHERE id = %s", (chair["stock"] - 1, chair_id))
```

Comment:
```suggestion
        cur.execute("UPDATE chair SET stock = stock - 1 WHERE id = %s", (chair_id,))
```
こういう風に実装していただけると助かります

Reply:
対応しました af5f92a